### PR TITLE
Added tests to cover link when appendProtocol is false

### PR DESCRIPTION
**Pull request by antoniopol06**

### CODE REVIEW

**Comment by jbalsas** [comment]
Hey @antoniopol06, [this](https://github.com/liferay/alloy-editor/pull/729/commits/466aefac048c9a24e9117c38a0bbefedbe671950) got me thinking... and I seem to remember that we did this on purpose, to allow the user to delete the protocol if he wished.

That was before we had the `appendProtocol` feature in place, so it might make sense to change this now, but I'm not 100% sure.

@ipeychev, what do you think?

**Comment by ipeychev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by ipeychev** [comment]
Hey @jbalsas,

The whole flow was intentional. On creating the link, we were adding the protocol, but in case of editing, we intentionally were leaving it as entered by the user. This was a heuristic approach, of course; we tried to guess what would be useful for the user and wanted to give him the last choice how the link should look like.

Since there is `appendProtocol` property however, I would say for consistence we have to respect it on editing too.

If I have to design it now, I would probably name it not `appendProtocol`, but `linkProtocol` (or simply `protocol` if it was part of `link` configuration and not a global one) and it would be not a boolean, but a string. The default value would be '' or null. Then, the user would be able to set 'http://', 'https://', '/' or something else.

In turn, the developer would be able to set the value to 'http', 'https' or '/' and not to `true`, as it is now, because this `true` doesn't say what would be the protocol of the link after all. However, since this is a part of the public API, we either have to add another property `linkProtocol` and deprecate the current one and later remove it, or to leave it and apply the protocol when we edit the link, as it is in this PR.

Thanks,

**Comment by jbalsas** [comment]
Hey @ipeychev, thanks a lot for the analysis, I totally agree with you!

I'll review and merge this as it is, and then we'll open a feature request to implement the `linkProtocol` API you drafted and deprecate this one.

Thanks1

File: test/core/test/link.js [hunk 1]
```diff
@@ -400,5 +400,96 @@
             assert.isTrue(range.endContainer.equals(range.endContainer));
             assert.strictEqual(range.endOffset, 1);
         });
+
+        describe('when appendProtocol is false', function() {
+
+            it('should not add default protocol when creating a link', function() {
+                if (CKEDITOR.env.ie) {
```
Comment (jbalsas) [inline test/core/test/link.js]: Hey @antoniopol06, just out of curiosity... is this test really failing in IE? Is the feature working? Same for the others below...

**Comment by antoniopol06** [comment]
Hey @jbalsas ,
You are right, I removed these lines of code from this PR.

**Comment by jbalsas** [comment]
merged, thanks!! :)

**Pull request closed without merging**